### PR TITLE
fix python_requires

### LINKED ISSUE #48: Invalid `python_requires` metadata stating the supported Python runtimes
One of the factors `pip install` uses to identify if a given version is compatible with the current Python runtime is [`python_requires` entry](https://setuptools.readthedocs.io/en/latest/setuptools.html#new-and-changed-setup-keywords). It is strongly recommended to add something like `python_requires='>=3'` whenever you drop support for Python 2 (same stands for other versions, of course).

If it's not set, pip will assume that the given version is compatible with Python 2 and will select it for installation regardless of whether the code in the dist is compatible with it. So this leads to having broken installs for users.

I noticed that you removed Python 2 support via https://github.com/sphinx-contrib/spelling/pull/33 and https://github.com/sphinx-contrib/spelling/pull/47/files. And you removed the corresponding [Trove classifier entry](https://pypi.org/help/#trove-classifier) but that's not enough.
Trove-classifiers are only used as tags/categories for search on PyPI. pip doesn't take them into account at all.

This is also visible at https://pypi.org/project/sphinxcontrib-spelling/5.1.0/ — there's no "Python requires" entry in the left column. For example, if you look at https://pypi.org/project/CherryPy/18.6.0/, you'll see that it requires Python `>= 3.5`.

How to fix this
=============

1. Add `python_requires` to your `setup.cfg`.
2. Cut a new release and verify that "Python requires" is present (note: PyPI reads this metadata from wheels, not sdists, so make sure to have them uploaded too)
3. Modern pip + modern Warehouse (PyPI) now implement support for ["yanking" releases] ([PEP 592]) and you should use it: just hit the "Yank" button on PyPI for the "broken" releases. This will make sure sure that those will not get `pip install`ed by accident (unless users will specify the exact version in their pins, as in `== 5.1.0`). It's better than deleting because it keeps the release available on PyPI (it will just have a special label on the UI).

[PEP 592]: https://www.python.org/dev/peps/pep-0592/
["yanking" releases]: https://pypi.org/help/#yanked

### PR DESCRIPTION
**Pull request by dhellmann**
Fixes #48

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -24,6 +24,13 @@ setenv =
 commands =
     flake8 sphinxcontrib setup.py
 
+[testenv:pkglint]
+deps=
+    twine
+commands=
+    python setup.py sdist
```
Comment (webknjaz) [inline tox.ini:31]: I strongly recommend to also lint wheels because their metadata is different. I'm not even sure if require-python is included there.

Comment (dhellmann) [reply tox.ini:31]: Hey, thanks, I didn't see your comment before I merged this so I'll look at that separately.

Comment (webknjaz) [reply tox.ini:31]: That's alright, I commented after the merge :)

Comment (dhellmann) [reply tox.ini:31]: Oh, good, I thought I hadn't been paying attention. :-) 

See #51 for the wheel linting.

**Pull request merged**